services/src/main/scala/org/apache/gearpump/services/AppMasterService.scala

```diff
@@ -51,13 +54,7 @@ trait AppMasterService extends HttpService  {
            case true =>
              onComplete((master ? AppMasterDataDetailRequest(appId)).asInstanceOf[Future[AppMasterDataDetail]]) {
                case Success(value: AppMasterDataDetail) =>
-                 if (isStreamingApplication(value)) {
-                   val streamApp: StreamingAppMasterDataDetail = value
-                   complete(write(streamApp))
-                 } else {
-                    val app: GeneralAppMasterDataDetail = value
-                   complete(write(app))
-                 }
+                 complete(value.toJson)
```

Comment:
Oh i see why toJson is used now. Can we only import upickle in ClusterMessage? That would nicely isolate JSON conversion to just messages which would allow us to change JSON serde easily.